src/portable/raspberrypi/rp2040/dcd_rp2040.c

```diff
@@ -164,9 +164,12 @@ static void _hw_endpoint_init(struct hw_endpoint *ep, uint8_t ep_addr, uint wMax
             ep->endpoint_control = &usb_dpram->ep_ctrl[num-1].out;
         }
 
-        // Now alloc a buffer and fill in endpoint control register
+        // Now if it hasn't already been done
+        //alloc a buffer and fill in endpoint control register
+        if( !(ep->configured)){
         _hw_endpoint_alloc(ep);
     }
```

Comment:
Looks like the indentation needs fixing?

Reply:
I agree. @ndinsmore can you fix the indentation and match the brace style of the rest of the file please. ie.

```
if(!(ep->configured))
{
     _hw_endpoint_alloc(ep);
}
```